### PR TITLE
feat(pwa): shared cache for nginx via memcached

### PR DESCRIPTION
**Pull request by dhhyi**
## PR Type

[x] Feature

## What Is the New Behavior?

PWA is deployed with a [memcached](https://memcached.org/) service used as a shared cache of the nginx servers.
This is a PoC open for discussion.

## Does this PR Introduce a Breaking Change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: charts/pwa/templates/deployment_memcached.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "pwa-memcached.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "memcached:latest"
```
Comment (khauser) [inline charts/pwa/templates/deployment_memcached.yaml:24]: Using latest is not best practice. Always try to use a specific version.

**Review by khauser — CHANGES_REQUESTED** [review]
At the moment the core-team replaces the webadapter pagecaching with a **redis-cache**. Already for that reason you should also use redis as caching system because we shouln't use different systems here for maintenance effort. From my knowledge redis does also have better features than memcached.
Is it possible to switch here to redis, or do you have some doubts?
Apart from that (as I also told the Core-Team) it's not **best practice** to define a deployment on our own. It's better use existing deployments (helm charts). You will find one also for memcached and can include it as a dependency.

@Thomas-Bergmann: What do you think here? In my opinion the PWA should use **the same** redis cache as icm-as for maintenance and resource reasons. In the end you would have two redis clusters for big customers.. that doesn't make sence.

**Comment by Thomas-Bergmann** [comment]
We discussed the options memcachd vs. redis, and the relations to the ICM. PWA is a separate application and it's not directly related to ICM. But for operational aspects, we decided to implement a "POC-" Redis connector too, so we can compare both solutions. If Redis matches the requirements, we will go with Redis. Additionally the cache invalidation must be part of the poc (and solution later).

**Comment by dhhyi** [comment]
replaced by #401

**Pull request closed without merging**